index.js

```diff
@@ -48,3 +23,34 @@ function updateUi() {
     canvas.getGrid().getOriginalTilesLength()
   )
 }
+
+// wire up callbacks
+function init() {
+  canvas.getGrid().onChange(() => updateUi())
+  ui.setAddTileCallback(id => canvas.getGrid().onAddTileMouseDown(id))
+  ui.setDatasetSelectedCallback(index => selectDataset(data.getDataset(index)))
+  ui.setCustomDatasetCallback(csv => selectDataset(data.parseCsv(csv)))
+  ui.setHightlightCallback(id => canvas.getGrid().onHighlightGeo(id))
+  ui.setUnhighlightCallback(() => canvas.getGrid().resetHighlightedGeo())
+  ui.setResolutionChangedCallback(value => {
+    hexagonGrid.setTileEdge(value)
+    canvas.updateTiles()
+  })
+  ui.setExportCallback(() => {
+    const json = exporter.formatTopoJson(canvas.getGrid().getTiles())
+    startDownload({
+      filename: 'hexagon-cartogram.json',
+      mimeType: 'application/json',
+      content: JSON.stringify(json),
+    })
+  })
+
+  // populate
+  ui.setGeos(mapData.getUniqueFeatureIds())
+  ui.setDatasetLabels(data.getLabels())
+  selectDataset(data.getDataset(0))
+  updateUi()
+}
+
+
```

Comment:
It's configurable! defaults to a max of 2 http://eslint.org/docs/rules/no-multiple-empty-lines but we could make it one!